Data_Structures_Questions.md

```diff
@@ -3,42 +3,46 @@ For each of the methods associated with each data structure, classify it based o
 ## Linked List
 
 1. What is the runtime complexity of `add_to_tail`?
-  
+    O(1)
+
     a. What if our list implementation didn't have a reference to the tail of the list in its constructor? What would be the runtime of the `add_to_tail` method?
+    O(N)
 
 2. What is the runtime complexity of `remove_head`?
-
+    O(1)
 3. What is the runtime complexity of `contains`?
-
+    O(N)
 4. What is the runtime complexity of `get_max`?
-
+    O(N)
 ## Queue
 
 1. What is the runtime complexity of `enqueue`?
-
+    O(1)
 2. What is the runtime complexity of `dequeue`?
-
+    O(1)
 3. What is the runtime complexity of `len`?
-
+    O(N)
 ## Binary Search Tree
 
 1. What is the runtime complexity of `insert`? 
-
+    O(N)
 2. What is the runtime complexity of `contains`?
-
+    O(2^N)
 3. What is the runtime complexity of `get_max`? 
+    O(N)
 
 ## Heap
 
 1. What is the runtime complexity of `_bubble_up`?
-
+    O(1)
 2. What is the runtime complexity of `_sift_down`?
-
+    O(1)
 3. What is the runtime complexity of `insert`?
-
+    O(N)
 4. What is the runtime complexity of `delete`?
-
+    O(2^N)
 5. What is the runtime complexity of `get_max`?
+    O(1)
 
```

Comment:
Some resources that might help with this: http://bigocheatsheet.com/
https://en.wikipedia.org/wiki/Heapsort